sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/signaling/TrouterUtils.java

```diff
@@ -365,23 +388,33 @@ private static BaseEvent getChatMessageReceived(JSONObject payload) {
         return eventPayload;
     }
 
+    private static String parseRecipientSkypeId(String skypeId) {
+        if (skypeId.startsWith(USER_PREFIX)) {
+            return skypeId;
+        }
+
+        return USER_PREFIX + skypeId;
+    }
+
     private static CommunicationIdentifier getUserIdentifier(String mri) {
-        if (mri.startsWith("8:orgid")) {
-            MicrosoftTeamsUserIdentifier userIdentifier = new MicrosoftTeamsUserIdentifier(mri, false);
-            userIdentifier.setCloudEnvironment(CommunicationCloudEnvironment.PUBLIC);
-            return userIdentifier;
-        } else if (mri.startsWith("8:dod")) {
-            MicrosoftTeamsUserIdentifier userIdentifier = new MicrosoftTeamsUserIdentifier(mri, false);
-            userIdentifier.setCloudEnvironment(CommunicationCloudEnvironment.DOD);
-            return userIdentifier;
-        } else if (mri.startsWith("8:gcch")) {
-            MicrosoftTeamsUserIdentifier userIdentifier = new MicrosoftTeamsUserIdentifier(mri, false);
-            userIdentifier.setCloudEnvironment(CommunicationCloudEnvironment.GCCH);
-            return userIdentifier;
-        } else if (mri.startsWith("8:teamsvisitor")) {
-            return new MicrosoftTeamsUserIdentifier(mri, true);
-        } else if (mri.startsWith("4:")) {
-            return new PhoneNumberIdentifier(mri);
+        if (mri.startsWith(TEAMS_PUBLIC_USER_PREFIX)) {
+            return new MicrosoftTeamsUserIdentifier(mri.substring(TEAMS_PUBLIC_USER_PREFIX.length()), false)
+                    .setRawId(mri)
+                    .setCloudEnvironment(CommunicationCloudEnvironment.PUBLIC);
+        } else if (mri.startsWith(TEAMS_DOD_USER_PREFIX)) {
+            return new MicrosoftTeamsUserIdentifier(mri.substring(TEAMS_DOD_USER_PREFIX.length()), false)
+                    .setRawId(mri)
+                    .setCloudEnvironment(CommunicationCloudEnvironment.DOD);
+        } else if (mri.startsWith(TEAMS_GCCH_USER_PREFIX)) {
+            return new MicrosoftTeamsUserIdentifier(mri.substring(TEAMS_GCCH_USER_PREFIX.length()), false)
+                    .setRawId(mri)
+                    .setCloudEnvironment(CommunicationCloudEnvironment.GCCH);
+        } else if (mri.startsWith(TEAMS_VISITOR_USER_PREFIX)) {
+            return new MicrosoftTeamsUserIdentifier(mri.substring(TEAMS_VISITOR_USER_PREFIX.length()), true)
+                .setRawId(mri);
+        } else if (mri.startsWith(PHONE_NUMBER_PREFIX)) {
+            return new PhoneNumberIdentifier(mri.substring(PHONE_NUMBER_PREFIX.length()))
+                .setRawId(mri);
         } else if (mri.startsWith("8:acs:") || mri.startsWith("8:spool:")) {
```

Comment:
Use constants above

Reply:
Ah... Good catch! I just forget to replace it...